frontend/src/app/features/enterprise/enterprise-trial.service.ts

```diff
@@ -7,44 +7,77 @@ import {
 import { PathHelperService } from 'core-app/core/path-helper/path-helper.service';
 import { ToastService } from 'core-app/shared/components/toaster/toast.service';
 import { FormGroup } from '@angular/forms';
-import { input } from 'reactivestates';
 import { EXTERNAL_REQUEST_HEADER } from 'core-app/features/hal/http/openproject-header-interceptor';
-
-export interface EnterpriseTrialData {
-  id?:string;
-  company:string;
-  first_name:string;
-  last_name:string;
-  email:string;
-  domain:string;
-  general_consent?:boolean;
-  newsletter_consent?:boolean;
-}
+import { EnterpriseTrialStore } from 'core-app/features/enterprise/enterprise-trial.store';
+import { GonType } from 'core-app/core/gon/gon.service';
+import {
+  EnterpriseTrialErrorHalResource,
+  EnterpriseTrialHalResource,
+  IEnterpriseData,
+  IEnterpriseTrial,
+} from 'core-app/features/enterprise/enterprise-trial.model';
+import { Query } from '@datorama/akita';
+import {
+  filter,
+  map,
+  shareReplay,
+} from 'rxjs/operators';
 
 @Injectable()
 export class EnterpriseTrialService {
-  // user data needs to be sync in ee-active-trial.component.ts
-  userData$ = input<EnterpriseTrialData>();
+  readonly store = new EnterpriseTrialStore();
 
-  public readonly baseUrlAugur:string;
+  private query = new Query(this.store);
 
-  public readonly tokenVersion:string;
+  confirmed$ = this.query.select('confirmed');
 
-  public trialLink:string;
+  cancelled$ = this.query.select('cancelled');
 
-  public resendLink:string;
+  status$ = this.query.select('status');
 
-  public modalOpen = false;
+  observe$ = this.query.select();
 
-  public confirmed:boolean;
+  userData$ = this.query
+    .select('data')
+    .pipe(
+      filter((data) => data !== undefined),
```

Comment:
there is a helper function, to not repeat this kind of filter again and again. you can write:

```
.pipe(filter(isDefinedEntity))
```

Reply:
replaced